checkov/main.py

```diff
@@ -201,6 +201,24 @@ def run(banner: str = checkov_banner, argv: List[str] = sys.argv[1:]) -> Optiona
                                  deep_analysis=config.deep_analysis,
                                  repo_root_for_plan_enrichment=config.repo_root_for_plan_enrichment)
 
+    source_env_val = os.getenv('BC_SOURCE', 'cli')
+    source = get_source_type(source_env_val)
+    if source == SourceTypes[BCSourceType.DISABLED]:
+        logger.warning(
+            f'Received unexpected value for BC_SOURCE: {source_env_val}; Should be one of {{{",".join(SourceTypes.keys())}}} setting source to DISABLED')
+    source_version = os.getenv('BC_SOURCE_VERSION', version)
+    logger.debug(f'BC_SOURCE = {source.name}, version = {source_version}')
+
+    if config.list:
+        # This speeds up execution by not setting up upload credentials (since we won't upload anything anyways)
+        logger.debug('Using --list; setting source to DISABLED')
+        source = SourceTypes[BCSourceType.DISABLED]
+
+    if CHECKOV_RUN_SCA_PACKAGE_SCAN_V2 and source.upload_results:
+        del DEFAULT_RUNNERS[-1]
+    else:
+        del DEFAULT_RUNNERS[-2]
```

Comment:
Rather than deleting the value from the list, perhaps add it here, e.g.
DEFAULT_RUNNERS.append(sca_package_runner())
reason is, in the future we might add more runners, and then your approach won't work because you'll delete the new runner